manifests/certonly.pp

```diff
@@ -48,6 +48,7 @@
 define letsencrypt::certonly (
   Enum['present','absent']                  $ensure               = 'present',
   Array[String[1]]                          $domains              = [$title],
+  Optional[String[1]]                       $cert_name            = $title,
```

Comment:
`Optional`? I think this should just be `String[1]`.  There's no way to set it to `undef` and that wouldn't be valid.